cli/src/commands/branch.rs

```diff
@@ -302,6 +307,11 @@ fn cmd_branch_rename(
     if view.get_local_branch(new_branch).is_present() {
         return Err(user_error(format!("Branch already exists: {new_branch}")));
     }
+    if has_tracked_remote_branches(view, new_branch) {
+        return Err(user_error(format!(
+            "Tracked remote branches exist for deleted branch: {new_branch}"
```

Comment:
If we decide to add something about `untrack` or `branch forget` in `jj branch delete`, we should probably add a hint about that here as well.